crates/noirc_frontend/src/hir/resolution/resolver.rs

```diff
@@ -674,6 +681,15 @@ impl<'a> Resolver<'a> {
         }
     }
 
+    /// True if the 'distinct' keyword is allowed on parameters in this function
+    fn distinct_allowed(&self, func: &NoirFunction) -> bool {
+        if self.in_contract() {
+            !func.def.is_unconstrained && !func.def.is_open
+        } else {
+            func.name() == "main"
```

Comment:
In the future, perhaps we should have a way to get the entry-point function instead of using "main"

Reply:
There may be a constant for `"main"` somewhere already in the name resolver

Reply:
hoisted